src/parse.rs

```diff
@@ -1523,8 +1517,8 @@ impl InlineStack {
         }
     }
 
-    fn pop_all<'a>(self, tree: &mut Tree<Item>) {
-        for el in self.stack {
+    fn pop_all<'a>(&mut self, tree: &mut Tree<Item>) {
+        for el in self.stack.drain(..) {
```

Comment:
I'm not sure why this works, but it doesn't break any tests. For future reference, the test the original version of the code was meant to fix is now [0.29 test 516](https://spec.commonmark.org/0.29/#example-516).

Reply:
The drain isn't functionally different than consuming the vector, right? (Note that the function signature moved from `self` to `&mut self`, so the (empty) vector inside can be reused later)